src/objects/domain.ts

```diff
@@ -40,4 +69,50 @@ export async function waitForDomain(): Promise<RunGraphDomain> {
   }
 
   return await waitForEvent('domainCreated')
+}
+
+async function watchDomain(domain: () => RunGraphDomain | undefined): Promise<void> {
+  const scope = await waitForScope()
+
+  scope.run(() => {
+    watch(domain, value => {
+      if (value) {
+        setDomain(value)
+      }
+    })
+  })
```

Comment:
Does this need to be torn down once the scope is disposed?

Reply:
The watcher? That's what the scope does. It's like I mounting a component.

Reply:
No worries! Scopes are pretty useful for this kind of stuff. The graph has a global scope it creates and automatically stops when the stop method is called. So in any other object like this we can use that scope as a lifecycle